include/SQLiteCpp/SQLiteCppExport.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * @file    SQLiteCppExport.h
+ * @ingroup SQLiteCpp
+ * @brief   File with macros needed in the generation of Windows DLLs
+ *
+ *
+ * Copyright (c) 2012-2022 Sebastien Rombauts (sebastien.rombauts@gmail.com)
+ *
+ * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
+ * or copy at http://opensource.org/licenses/MIT)
+ */
+
+#pragma once
+
+/*
+* #define SQLITECPP_COMPILE_DLL to compile a DLL under Windows
+* #define SQLITECPP_EXPORT to export symbols when creating the DLL, otherwise it defaults to importing symbols
+*/
+
+/* Windows DLL export/import */
+#if defined(WIN32) && defined(SQLITECPP_COMPILE_DLL)
+    #if SQLITECPP_DLL_EXPORT
+        #define DLL_API __declspec(dllexport)
+        #pragma message("Exporting symbols")
+    #else
+        #define DLL_API __declspec(dllimport)
+        #pragma message("Importing symbols")
+    #endif    
+#else    
+    #define DLL_API
```

Comment:
You can also check if the compiler supports GNUC, and set up:

```c
__attribute__ ((visibility ("default")))
```

This then allows  you to specify on unix that the software should be compiled with e.g. meson's `gnu_symbol_visibility: 'hidden'`, which allows avoiding accidental leakage of private symbols, and also allows better codegen.

Reply:
Done, thanks for the suggestion